File: data/recomposed_projects/llama-3-70b-instruct/body/joda-money/evosuite-test/src/test/org/joda/money/format/MultiPrinterParser_ESTest.py
from __future__ import annotations
import time
import locale
import re
import mock
import os
import decimal
from io import StringIO
import unittest
import pytest
import io
import unittest

# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *
from src.main.org.joda.money.BigMoney import *
from src.main.org.joda.money.CurrencyUnit import *
from src.main.org.joda.money.format.AmountPrinterParser import *
from src.main.org.joda.money.format.LiteralPrinterParser import *
from src.main.org.joda.money.format.MoneyAmountStyle import *
from src.main.org.joda.money.format.MoneyFormatter import *
from src.main.org.joda.money.format.MoneyFormatterBuilder import *
from src.main.org.joda.money.format.MoneyParseContext import *
from src.main.org.joda.money.format.MoneyParser import *
from src.main.org.joda.money.format.MoneyPrintContext import *
from src.main.org.joda.money.format.MoneyPrinter import *
from src.main.org.joda.money.format.MultiPrinterParser import *

# from src.test.org.joda.money.format.MultiPrinterParser_ESTest_scaffolding import *


class MultiPrinterParser_ESTest(unittest.TestCase):

    def test21(self) -> None:

        money_printer_array0 = [None]
        money_parser_array0 = [None, None]
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        locale0 = "de"
        currency_unit0 = CurrencyUnit.GBP
        big_decimal0 = 10
        money_parse_context0 = MoneyParseContext(
            8, "tRyAM_&cl9t3j", (-10), locale0, big_decimal0, (-10), currency_unit0
        )

        try:
            multi_printer_parser0.parse(money_parse_context0)
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            self.verifyException("org.joda.money.format.MultiPrinterParser", e)

    def test20(self) -> None:

        literalPrinterParser0 = LiteralPrinterParser("Bx")
        moneyPrinterArray0 = [None] * 1
        moneyPrinterArray0[0] = literalPrinterParser0
        moneyParserArray0 = [None] * 2
        multiPrinterParser0 = MultiPrinterParser(moneyPrinterArray0, moneyParserArray0)
        string0 = multiPrinterParser0.toString()
        self.assertIsNotNone(string0)

    def test19(self) -> None:

        money_printer_array0 = [MoneyPrinter()]
        money_formatter_builder0 = MoneyFormatterBuilder()
        money_parser_array0 = [MoneyParser(), MoneyParser()]
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        multi_printer_parser0.appendTo(money_formatter_builder0)

    def test18(self) -> None:

        money_printer_array0 = []
        money_parser_array0 = []
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        locale0 = Locale.US
        money_parser_array1 = [None] * 2
        money_parser_array1[0] = multi_printer_parser0
        money_parser_array1[1] = multi_printer_parser0
        multi_printer_parser1 = MultiPrinterParser(
            money_printer_array0, money_parser_array1
        )
        money_formatter0 = MoneyFormatter(
            1, locale0, money_parser_array0, money_printer_array0, multi_printer_parser1
        )
        money_parse_context0 = money_formatter0.parse("V$_G", 1)
        self.assertFalse(money_parse_context0.isError())

    def test17(self) -> None:

        money_printer_array0 = [None] * 1
        literal_printer_parser0 = LiteralPrinterParser("Bx")
        money_parser_array0 = [None] * 2
        money_parser_array0[0] = literal_printer_parser0
        money_parser_array0[1] = literal_printer_parser0
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        locale0 = Locale.FRENCH
        money_formatter0 = MoneyFormatter(
            (-4158),
            locale0,
            money_parser_array0,
            money_printer_array0,
            multi_printer_parser0,
        )
        money_parse_context0 = money_formatter0.parse(
            "org.joda.money.format.LiteralPrinterParser@0000000001org.joda.money.format.LiteralPrinterParser@0000000001",
            1,
        )
        multi_printer_parser0.parse(money_parse_context0)
        self.assertEqual(1, money_parse_context0.getErrorIndex())
        self.assertTrue(money_parse_context0.isError())

    def test16(self) -> None:

        money_printer_array0 = [None] * 1
        literal_printer_parser0 = LiteralPrinterParser("Bx")
        money_parser_array0 = [None] * 2
        money_parser_array0[0] = literal_printer_parser0
        money_parser_array0[1] = literal_printer_parser0
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        string0 = multi_printer_parser0.toString()
        self.assertIsNotNone(string0)

    def test15(self) -> None:

        literalPrinterParser0 = LiteralPrinterParser("Bx")
        moneyPrinterArray0 = [None] * 1
        moneyPrinterArray0[0] = literalPrinterParser0
        moneyParserArray0 = [None] * 2
        moneyParserArray0[0] = literalPrinterParser0
        moneyParserArray0[1] = literalPrinterParser0
        multiPrinterParser0 = MultiPrinterParser(moneyPrinterArray0, moneyParserArray0)
        string0 = multiPrinterParser0.toString()
        self.assertIsNotNone(string0)

    def test14(self) -> None:

        money_printer_array0 = [None] * 8
        money_parser_array0 = [None] * 1
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        string0 = multi_printer_parser0.toString()
        self.assertEqual("", string0)

    def test13(self) -> None:

        money_printer_array0 = [None] * 5
        money_parser_array0 = []
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        money_formatter_builder0 = MoneyFormatterBuilder()

        try:
            multi_printer_parser0.appendTo(money_formatter_builder0)
            self.fail("Expecting exception: ArrayIndexOutOfBoundsException")

        except IndexError as e:
            self.assertEqual(str(e), "list index out of range")

    def test12(self) -> None:

        money_printer_array0 = [None]
        multi_printer_parser0 = MultiPrinterParser(money_printer_array0, None)
        money_formatter_builder0 = MoneyFormatterBuilder()

        try:
            multi_printer_parser0.appendTo(money_formatter_builder0)
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            self.verifyException("org.joda.money.format.MultiPrinterParser", e)

    def test11(self) -> None:
        moneyPrinterArray0 = []
        multiPrinterParser0 = MultiPrinterParser(moneyPrinterArray0, None)
        with pytest.raises(RuntimeError):
            multiPrinterParser0.isParser()

    def test10(self) -> None:
        multiPrinterParser0 = MultiPrinterParser(None, None)
        try:
            multiPrinterParser0.isPrinter()
            self.fail("Expecting exception: RuntimeError")
        except Exception as e:
            self.assertEqual(str(e), "")

    def test09(self) -> None:

        money_printer_array0 = [None] * 2
        money_amount_style0 = MoneyAmountStyle.ASCII_DECIMAL_POINT_GROUP3_COMMA
        amount_printer_parser0 = AmountPrinterParser(money_amount_style0)
        money_parser_array0 = [None] * 1
        money_parser_array0[0] = amount_printer_parser0
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        big_decimal0 = decimal.Decimal(10)
        money_parse_context0 = MoneyParseContext(
            5, "£", 2304, None, big_decimal0, 2304, None
        )

        try:
            multi_printer_parser0.parse(money_parse_context0)
            self.fail("Expecting exception: RuntimeError")
        except Exception as e:
            self.assertIsInstance(e, RuntimeError)
            self.verifyException("org.joda.money.format.MoneyFormatter", e)

    def test08(self) -> None:

        money_printer_array0 = [None] * 1
        literal_printer_parser0 = LiteralPrinterParser("Bx")
        money_parser_array0 = [None] * 2
        money_parser_array0[0] = literal_printer_parser0
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        locale0 = "de"
        currency_unit0 = CurrencyUnit.GBP
        big_decimal0 = 10
        money_parse_context0 = MoneyParseContext(
            8, "tRyAM_&cl9t3j", -10, locale0, big_decimal0, -10, currency_unit0
        )

        try:
            multi_printer_parser0.parse(money_parse_context0)
            self.fail("Expecting exception: StringIndexOutOfBoundsException")
        except StringIndexOutOfBoundsException:
            pass

    def test07(self) -> None:

        money_printer_array0 = [None] * 4
        money_parser_array0 = []
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        locale0 = Locale.US
        money_print_context0 = MoneyPrintContext(locale0)
        mock_file0 = MockFile("^23s*N:l2R.#[,)K%'")
        mock_file_writer0 = MockFileWriter(mock_file0, True)

        try:
            multi_printer_parser0.print(money_print_context0, mock_file_writer0, None)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.joda.money.format.MultiPrinterParser", e)

    def test06(self) -> None:

        literalPrinterParser0 = LiteralPrinterParser("Bx")
        moneyPrinterArray0 = [literalPrinterParser0]
        moneyParserArray0 = []
        charBuffer0 = io.StringIO("Money iterator must not contain n~ll entriQs")
        multiPrinterParser0 = MultiPrinterParser(moneyPrinterArray0, moneyParserArray0)

        try:
            multiPrinterParser0.print(None, charBuffer0, None)
            self.fail("Expecting exception: ReadOnlyBufferException")

        except Exception as e:
            self.verifyException("java.nio.CharBuffer", e)

    def test05(self) -> None:

        money_printer_array0 = [None]
        multi_printer_parser0 = MultiPrinterParser(money_printer_array0, None)

        try:
            multi_printer_parser0.toString()
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            self.verifyException("java.util.Objects", e)

    def test04(self) -> None:
        moneyParserArray0 = [None] * 3
        moneyPrinterArray0 = [None] * 7
        multiPrinterParser0 = MultiPrinterParser(moneyPrinterArray0, moneyParserArray0)
        boolean0 = multiPrinterParser0.isParser()
        self.assertFalse(boolean0)

    def test03(self) -> None:

        money_printer_array0 = []
        money_parser_array0 = [None] * 1
        literal_printer_parser0 = LiteralPrinterParser("")
        money_parser_array0[0] = literal_printer_parser0
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        boolean0 = multi_printer_parser0.isParser()
        self.assertTrue(boolean0)

    def test02(self) -> None:
        moneyPrinterArray0 = [None] * 3
        moneyParserArray0 = [None] * 9
        multiPrinterParser0 = MultiPrinterParser(moneyPrinterArray0, moneyParserArray0)
        boolean0 = multiPrinterParser0.isPrinter()
        self.assertFalse(boolean0)

    def test01(self) -> None:

        money_printer_array0 = [None] * 2
        money_amount_style0 = MoneyAmountStyle.ASCII_DECIMAL_POINT_GROUP3_COMMA
        amount_printer_parser0 = AmountPrinterParser(money_amount_style0)
        money_printer_array0[0] = amount_printer_parser0
        money_printer_array0[1] = amount_printer_parser0
        money_parser_array0 = [None] * 1
        multi_printer_parser0 = MultiPrinterParser(
            money_printer_array0, money_parser_array0
        )
        boolean0 = multi_printer_parser0.isPrinter()
        self.assertTrue(boolean0)

    def test00(self) -> None:

        locale0 = Locale("7<9]ug;4-LO2Sd(ns")
        moneyPrintContext0 = MoneyPrintContext(locale0)
        mockFileWriter0 = MockFileWriter("7<9]ug;4-LO2Sd(ns")
        moneyPrinterArray0 = []
        multiPrinterParser0 = MultiPrinterParser(moneyPrinterArray0, None)
        multiPrinterParser0.print(moneyPrintContext0, mockFileWriter0, None)

File: java_projects/cleaned_final_projects_evosuite/joda-money/evosuite-tests/org/joda/money/format/MultiPrinterParser_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.joda.money.format;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.math.BigDecimal;
import java.nio.CharBuffer;
import java.nio.ReadOnlyBufferException;
import java.util.Locale;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.mock.java.io.MockFile;
import org.evosuite.runtime.mock.java.io.MockFileWriter;
import org.joda.money.BigMoney;
import org.joda.money.CurrencyUnit;
import org.joda.money.format.AmountPrinterParser;
import org.joda.money.format.LiteralPrinterParser;
import org.joda.money.format.MoneyAmountStyle;
import org.joda.money.format.MoneyFormatter;
import org.joda.money.format.MoneyFormatterBuilder;
import org.joda.money.format.MoneyParseContext;
import org.joda.money.format.MoneyParser;
import org.joda.money.format.MoneyPrintContext;
import org.joda.money.format.MoneyPrinter;
import org.joda.money.format.MultiPrinterParser;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class MultiPrinterParser_ESTest extends MultiPrinterParser_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      Locale locale0 = new Locale("7<9]ug;4-LO2Sd(ns");
      MoneyPrintContext moneyPrintContext0 = new MoneyPrintContext(locale0);
      MockFileWriter mockFileWriter0 = new MockFileWriter("7<9]ug;4-LO2Sd(ns");
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[0];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, (MoneyParser[]) null);
      multiPrinterParser0.print(moneyPrintContext0, mockFileWriter0, (BigMoney) null);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[2];
      MoneyAmountStyle moneyAmountStyle0 = MoneyAmountStyle.ASCII_DECIMAL_POINT_GROUP3_COMMA;
      AmountPrinterParser amountPrinterParser0 = new AmountPrinterParser(moneyAmountStyle0);
      moneyPrinterArray0[0] = (MoneyPrinter) amountPrinterParser0;
      moneyPrinterArray0[1] = (MoneyPrinter) amountPrinterParser0;
      MoneyParser[] moneyParserArray0 = new MoneyParser[1];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      boolean boolean0 = multiPrinterParser0.isPrinter();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[3];
      MoneyParser[] moneyParserArray0 = new MoneyParser[9];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      boolean boolean0 = multiPrinterParser0.isPrinter();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[0];
      MoneyParser[] moneyParserArray0 = new MoneyParser[1];
      LiteralPrinterParser literalPrinterParser0 = new LiteralPrinterParser("");
      moneyParserArray0[0] = (MoneyParser) literalPrinterParser0;
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      boolean boolean0 = multiPrinterParser0.isParser();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      MoneyParser[] moneyParserArray0 = new MoneyParser[3];
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[7];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      boolean boolean0 = multiPrinterParser0.isParser();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, (MoneyParser[]) null);
      // Undeclared exception!
      try { 
        multiPrinterParser0.toString();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.util.Objects", e);
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      LiteralPrinterParser literalPrinterParser0 = new LiteralPrinterParser("Bx");
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      moneyPrinterArray0[0] = (MoneyPrinter) literalPrinterParser0;
      MoneyParser[] moneyParserArray0 = new MoneyParser[1];
      CharBuffer charBuffer0 = CharBuffer.wrap((CharSequence) "Money iterator must not contain n~ll entriQs");
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      // Undeclared exception!
      try { 
        multiPrinterParser0.print((MoneyPrintContext) null, charBuffer0, (BigMoney) null);
        fail("Expecting exception: ReadOnlyBufferException");
      
      } catch(ReadOnlyBufferException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.nio.CharBuffer", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[4];
      MoneyParser[] moneyParserArray0 = new MoneyParser[0];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      Locale locale0 = Locale.US;
      MoneyPrintContext moneyPrintContext0 = new MoneyPrintContext(locale0);
      MockFile mockFile0 = new MockFile("^23s*N:l2R.#[,)K%'");
      MockFileWriter mockFileWriter0 = new MockFileWriter(mockFile0, true);
      // Undeclared exception!
      try { 
        multiPrinterParser0.print(moneyPrintContext0, mockFileWriter0, (BigMoney) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.joda.money.format.MultiPrinterParser", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      LiteralPrinterParser literalPrinterParser0 = new LiteralPrinterParser("Bx");
      MoneyParser[] moneyParserArray0 = new MoneyParser[2];
      moneyParserArray0[0] = (MoneyParser) literalPrinterParser0;
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      Locale locale0 = Locale.GERMAN;
      CurrencyUnit currencyUnit0 = CurrencyUnit.GBP;
      BigDecimal bigDecimal0 = BigDecimal.TEN;
      MoneyParseContext moneyParseContext0 = new MoneyParseContext(8, "tRyAM_&cl9t3j", (-10), locale0, bigDecimal0, (-10), currencyUnit0);
      // Undeclared exception!
      try { 
        multiPrinterParser0.parse(moneyParseContext0);
        fail("Expecting exception: StringIndexOutOfBoundsException");
      
      } catch(StringIndexOutOfBoundsException e) {
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[2];
      MoneyAmountStyle moneyAmountStyle0 = MoneyAmountStyle.ASCII_DECIMAL_POINT_GROUP3_COMMA;
      AmountPrinterParser amountPrinterParser0 = new AmountPrinterParser(moneyAmountStyle0);
      MoneyParser[] moneyParserArray0 = new MoneyParser[1];
      moneyParserArray0[0] = (MoneyParser) amountPrinterParser0;
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      BigDecimal bigDecimal0 = BigDecimal.TEN;
      MoneyParseContext moneyParseContext0 = new MoneyParseContext(5, "\u00A3", 2304, (Locale) null, bigDecimal0, 2304, (CurrencyUnit) null);
      // Undeclared exception!
      try { 
        multiPrinterParser0.parse(moneyParseContext0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // Locale must not be null
         //
         verifyException("org.joda.money.format.MoneyFormatter", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser((MoneyPrinter[]) null, (MoneyParser[]) null);
      // Undeclared exception!
      try { 
        multiPrinterParser0.isPrinter();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.util.Objects", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[0];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, (MoneyParser[]) null);
      // Undeclared exception!
      try { 
        multiPrinterParser0.isParser();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.util.Objects", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, (MoneyParser[]) null);
      MoneyFormatterBuilder moneyFormatterBuilder0 = new MoneyFormatterBuilder();
      // Undeclared exception!
      try { 
        multiPrinterParser0.appendTo(moneyFormatterBuilder0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.joda.money.format.MultiPrinterParser", e);
      }
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[5];
      MoneyParser[] moneyParserArray0 = new MoneyParser[0];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      MoneyFormatterBuilder moneyFormatterBuilder0 = new MoneyFormatterBuilder();
      // Undeclared exception!
      try { 
        multiPrinterParser0.appendTo(moneyFormatterBuilder0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // Index 0 out of bounds for length 0
         //
         verifyException("org.joda.money.format.MultiPrinterParser", e);
      }
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[8];
      MoneyParser[] moneyParserArray0 = new MoneyParser[1];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      String string0 = multiPrinterParser0.toString();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      LiteralPrinterParser literalPrinterParser0 = new LiteralPrinterParser("Bx");
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      moneyPrinterArray0[0] = (MoneyPrinter) literalPrinterParser0;
      MoneyParser[] moneyParserArray0 = new MoneyParser[2];
      moneyParserArray0[0] = (MoneyParser) literalPrinterParser0;
      moneyParserArray0[1] = (MoneyParser) literalPrinterParser0;
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      String string0 = multiPrinterParser0.toString();
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      LiteralPrinterParser literalPrinterParser0 = new LiteralPrinterParser("Bx");
      MoneyParser[] moneyParserArray0 = new MoneyParser[2];
      moneyParserArray0[0] = (MoneyParser) literalPrinterParser0;
      moneyParserArray0[1] = (MoneyParser) literalPrinterParser0;
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      String string0 = multiPrinterParser0.toString();
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      LiteralPrinterParser literalPrinterParser0 = new LiteralPrinterParser("Bx");
      MoneyParser[] moneyParserArray0 = new MoneyParser[2];
      moneyParserArray0[0] = (MoneyParser) literalPrinterParser0;
      moneyParserArray0[1] = (MoneyParser) literalPrinterParser0;
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      Locale locale0 = Locale.FRENCH;
      MoneyFormatter moneyFormatter0 = new MoneyFormatter((-4158), locale0, moneyParserArray0, moneyPrinterArray0, multiPrinterParser0);
      MoneyParseContext moneyParseContext0 = moneyFormatter0.parse("org.joda.money.format.LiteralPrinterParser@0000000001org.joda.money.format.LiteralPrinterParser@0000000001", 1);
      multiPrinterParser0.parse(moneyParseContext0);
      assertEquals(1, moneyParseContext0.getErrorIndex());
      assertTrue(moneyParseContext0.isError());
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[0];
      MoneyParser[] moneyParserArray0 = new MoneyParser[0];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      Locale locale0 = Locale.US;
      MoneyParser[] moneyParserArray1 = new MoneyParser[2];
      moneyParserArray1[0] = (MoneyParser) multiPrinterParser0;
      moneyParserArray1[1] = (MoneyParser) multiPrinterParser0;
      MultiPrinterParser multiPrinterParser1 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray1);
      MoneyFormatter moneyFormatter0 = new MoneyFormatter(1, locale0, moneyParserArray0, moneyPrinterArray0, multiPrinterParser1);
      MoneyParseContext moneyParseContext0 = moneyFormatter0.parse("V$_G", 1);
      assertFalse(moneyParseContext0.isError());
  }

  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      MoneyFormatterBuilder moneyFormatterBuilder0 = new MoneyFormatterBuilder();
      MoneyParser[] moneyParserArray0 = new MoneyParser[2];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      multiPrinterParser0.appendTo(moneyFormatterBuilder0);
  }

  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      LiteralPrinterParser literalPrinterParser0 = new LiteralPrinterParser("Bx");
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      moneyPrinterArray0[0] = (MoneyPrinter) literalPrinterParser0;
      MoneyParser[] moneyParserArray0 = new MoneyParser[2];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      String string0 = multiPrinterParser0.toString();
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test21()  throws Throwable  {
      MoneyPrinter[] moneyPrinterArray0 = new MoneyPrinter[1];
      MoneyParser[] moneyParserArray0 = new MoneyParser[2];
      MultiPrinterParser multiPrinterParser0 = new MultiPrinterParser(moneyPrinterArray0, moneyParserArray0);
      Locale locale0 = Locale.GERMAN;
      CurrencyUnit currencyUnit0 = CurrencyUnit.GBP;
      BigDecimal bigDecimal0 = BigDecimal.TEN;
      MoneyParseContext moneyParseContext0 = new MoneyParseContext(8, "tRyAM_&cl9t3j", (-10), locale0, bigDecimal0, (-10), currencyUnit0);
      // Undeclared exception!
      try { 
        multiPrinterParser0.parse(moneyParseContext0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.joda.money.format.MultiPrinterParser", e);
      }
  }
}
